### PR TITLE
Adds way to reevaluate the access of a node in the admin UI

### PR DESCRIPTION
**Pull request by julsemaan**

### CODE REVIEW

**Comment by julsemaan** [comment]
can we merge ?

File: html/pfappserver/lib/pfappserver/Controller/Node.pm [hunk 1]
```diff
@@ -319,6 +320,16 @@ sub delete :Chained('object') :PathPart('delete') :Args(0) :AdminRole('NODES_DEL
     $c->stash->{current_view} = 'JSON';
 }
 
+sub reevaluate_access :Chained('object') :PathPart('reevaluate_access') :Args(0) :AdminRole('NODES_UPDATE') {
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Controller/Node.pm:329]: Add a pod header with a meaningful description.

File: html/pfappserver/lib/pfappserver/Controller/Node.pm [hunk 2]
```diff
@@ -319,6 +320,16 @@ sub delete :Chained('object') :PathPart('delete') :Args(0) :AdminRole('NODES_DEL
...
+    my ( $self, $c ) = @_;
+
+    my ($status, $message) = $c->model('Node')->reevaluate($c->stash->{mac});
+    $c->response->status($status);
+    $c->stash->{status_msg} = $message; # TODO: localize error message
+    $c->stash->{current_view} = 'JSON';
+
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Controller/Node.pm]: Remove this empty line.

File: html/pfappserver/lib/pfappserver/Model/Node.pm [hunk 3]
```diff
@@ -751,6 +768,21 @@ sub bulkApplyRole {
...
 
+=head2 bulkReevaluateAccess
+
+=cut
+
+sub bulkReevaluateAccess {
+    my ($self, $role, @macs) = @_;
+    my $count = 0;
+    foreach my $mac (@macs) {
+        if(reevaluate_access($mac, "node_modify")){
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Model/Node.pm]: Please be consistent with whitespaces. This line should be:
`if (reevaluate_access($mac, "node_modify")) {`

File: html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm [hunk 4]
```diff
@@ -120,6 +120,28 @@ sub bulk_apply_violation : Local : Args(1) {
     $c->stash( status_msg => $status_msg, );
 }
 
+=head2 bulk_reevaluate_access
+
+=cut
+
+sub bulk_reevaluate_access : Local {
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm:127]: Please be consistent with whitespaces. This sub is having spaces all over the place.

File: html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm [hunk 5]
```diff
@@ -120,6 +120,28 @@ sub bulk_apply_violation : Local : Args(1) {
...
+    my ( $self, $c ) = @_;
+    $c->stash->{current_view} = 'JSON';
+    my ( $status, $status_msg );
+    my $request = $c->request;
+    if ( $request->method eq 'POST' ) {
+        my @ids = $request->param('items');
+        ( $status, $status_msg ) =
+          $self->getModel($c)->bulkReevaluateAccess( @ids );
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm]: Please don't split this assignation as it doesn't take more than 120 characters.

File: html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm (hunk 4 above)
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm:127]: I copy pasted the sub above it. If we fix this one, we need to refactor the whole module for consistency...

File: html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm (hunk 5 above)
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm]: Same as above here..

File: html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm (hunk 4 above)
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm:127]: Please take this opportunity to improve the code. Why would you repeat the mistakes of others?

Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Role/Controller/BulkActions.pm:127]: Done, the sub is corrected. Once the PR is merged I'll refactor the other subs in devel.

**Pull request merged**